### ISSUE TITLE
G4mespeed mod

### ISSUE THREAD

**Issue by ardissaps** [ZsoltMolnarrr/BetterCombat#10]
https://paste.atlauncher.com/view/3dfd2c89-d0f5-48ef-b7f1-de23f095c465

**Comment by floral-qua-floral** [comment]
Was able to confirm that this is definitely an incompatibility between Better Combat and G4mespeed. So it might be worth changing the title of the issue to "Crash on startup with g4mespeed mod" or something similar. Here's a crash report where only Fabric API, CompleteConfig, Better Combat, and G4mespeed are loaded:
[crash-2022-07-20_18.53.01-client.txt](https://github.com/ZsoltMolnarrr/BetterCombat/files/9155529/crash-2022-07-20_18.53.01-client.txt)

I'm already in the G4mespeed mod's discord server, so I'll put a message in the support channel there linking to this issue so that that mod's developer can be aware of the incompatibility as well.

**Comment by ardissaps** [comment]
I tried disabled g4merspeed today but it still won't launch to the game. Seems like conflict with another mods too?
https://paste.atlauncher.com/view/1ce84704-88b7-49b9-8083-23a495912476

**Comment by chronosacaria** [comment]
@ardissaps This new issue that you are presenting appears to be between BetterCombat and Custom Crosshair Mod as per this part of your log (again, mapped for ease of reading):
```
[21/07/2022 09:39:32 AM] [Render thread/WARN] Configuration conflict: there is more than one oshi.architecture.properties file on the classpath
[21/07/2022 09:39:32 AM] ---- Minecraft Crash Report ----
[21/07/2022 09:39:32 AM] // Hi. I'm Minecraft, and I'm a crashaholic.
[21/07/2022 09:39:32 AM] 
[21/07/2022 09:39:32 AM] Time: 7/21/22, 9:39 AM
[21/07/2022 09:39:32 AM] Description: Initializing game
[21/07/2022 09:39:32 AM] 
[21/07/2022 09:39:32 AM] java.lang.RuntimeException: Mixin transformation of net.minecraft.client.gui.hud.InGameHud failed
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:419)
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145)
[21/07/2022 09:39:32 AM] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
[21/07/2022 09:39:32 AM] 	at net.minecraft.client.MinecraftClient.<init>(MinecraftClient.java:593)
[21/07/2022 09:39:32 AM] 	at net.minecraft.client.main.Main.main(Main.java:197)
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
[21/07/2022 09:39:32 AM] Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
[21/07/2022 09:39:32 AM] 	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414)
[21/07/2022 09:39:32 AM] 	... 9 more
[21/07/2022 09:39:32 AM] Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Redirector crosshairmod$renderCrosshair(Lnet/minecraft/InGameHud;Lnet/minecraft/MatrixStack;IIIIII)V in crosshairmod.mixins.json:InGameHudMixin from mod crosshairmod failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap crosshairmod-refmap.json
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
[21/07/2022 09:39:32 AM] 	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
[21/07/2022 09:39:32 AM] 	... 12 more
```
It looks like they are both trying to target `Lnet/minecraft/InGameHud;Lnet/minecraft/MatrixStack;IIIIII)V`; Better Combat via `InGameHudInject#pre_renderCrosshair` and Custom Crosshair Mod via `crosshairmod$renderCrosshair`. It also appears that the Custom Crosshair Mod is trying to `@Redirect` as well, like BetterCombat is. 

As Custom Crosshair Mod is, as G4mespeed is, All Rights Reserved and not Open Source, I cannot see what, exactly, is going on. However, it would probably be advisable to make a separate issue concerning this crash as it is related to a different mod. I would suggest a title, something along the lines of: "Crash on startup with Custom Crosshair Mod (v. 0.4.5)". You could also either copy and paste the mapped crash report or just link to this comment to make things easier for yourself

**Comment by ZsoltMolnarrr** [comment]
Since that mod is not open source, priority for fixing is super low.

**Comment by G4me4u** [comment]
Hello! I am guessing you already fixed the issue in 4de01e6. I would have to actually check it myself though.

I was on vacation when seeing this issue, so I had no way to resolve it myself.

**Comment by ZsoltMolnarrr** [comment]
Seems to be fixed.
